### PR TITLE
Create VolRep CRs for valid PVCs from VRG

### PR DESCRIPTION
**Pull request by ShyamsundarR**
Update VRG reconcier to create VolumeReplication CR
for PVCs that match a VRG CR.

Signed-off-by: Shyamsundar Ranganathan <srangana@redhat.com>
Sidned-off-by: Priyaranjan Behera <pbehera@redhat.com>

### CODE REVIEW

**Review by raghavendrabhat — APPROVED** [review]
LGTM

**Review by BenamarMk — COMMENTED** [review]
Done

File: controllers/volumereplicationgroup_controller.go [hunk 1]
```diff
@@ -197,19 +201,34 @@ func (v *VolumeReplicationGroupReconciler) Reconcile(ctx context.Context, req ct
...
+		err = v.Get(ctx, types.NamespacedName{Name: pvc.Name, Namespace: pvc.Namespace}, volRep)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				if err = v.createVolumeReplicationCRForPVC(ctx, pvc); err == nil {
+					continue
+				}
+			}
+
+			// requeue on failure to ensure any PVC not having a corresponding VR CR
+			requeue = true
```
Comment (ShyamsundarR) [inline controllers/volumereplicationgroup_controller.go:223]: Success of get and create was tested, error condition remains untested to ensure re-queue.

File: controllers/volumereplicationgroup_controller.go [hunk 2]
```diff
@@ -234,7 +253,7 @@ func (v *VolumeReplicationGroupReconciler) HandlePersistentVolumeClaims(
 		return pvcList, fmt.Errorf("failed to list PVC %w", err)
 	}
 
-	log.Info("PVC List: ", pvcList)
+	log.Info("Found PVCs, count: ", len(pvcList.Items))
```
Comment (ShyamsundarR) [inline controllers/volumereplicationgroup_controller.go:256]: Reduced logging content from all PVCs to the count.

File: controllers/volumereplicationgroup_controller.go [hunk 3]
```diff
@@ -202,11 +202,29 @@ func (v *VolumeReplicationGroupReconciler) Reconcile(ctx context.Context, req ct
...
+			}
+		} else if err != nil {
+			log.Error(err, "Failed to get VolumeReplication CR")
+
+			return ctrl.Result{}, fmt.Errorf("failed to get CR: %w", err)
+		}
+	}
+
+	// Requeue if new CR is created
+	if requeue {
```
[the remark this answers is not in the document]
Comment (ShyamsundarR) [reply controllers/volumereplicationgroup_controller.go]: Updated requeue decision.

File: controllers/volumereplicationgroup_controller.go (hunk 1 above)
Comment (ShyamsundarR) [inline controllers/volumereplicationgroup_controller.go:223]: I wanted these statements cuddled, IOW like,
```
			// requeue on failure to ensure any PVC not having a corresponding VR CR
			requeue = true
			log.Error(err, "Failed to get or create VolumeReplication CR for PVC", pvc.Name, pvc.Namespace)

			continue
```

This is the [rule](https://github.com/bombsimon/wsl/blob/master/doc/configuration.md#allow-assign-and-call) that disallows it.

Wanted to check if this rule in the `wsl` linter can be disabled, as it seems too strict and IMO causes a lot of line spaces in the code.

File: controllers/volumereplicationgroup_controller.go [hunk 4]
```diff
@@ -197,19 +201,34 @@ func (v *VolumeReplicationGroupReconciler) Reconcile(ctx context.Context, req ct
...
+					continue
+				}
+			}
+
+			// requeue on failure to ensure any PVC not having a corresponding VR CR
+			requeue = true
+
+			log.Error(err, "Failed to get or create VolumeReplication CR for PVC", pvc.Name, pvc.Namespace)
+
+			continue
```
Comment (BenamarMk) [inline controllers/volumereplicationgroup_controller.go:227]: `continue` not needed

Comment (ShyamsundarR) [reply controllers/volumereplicationgroup_controller.go]: Ugh! yes addressed...

**Review by BenamarMk — APPROVED** [review]

**Comment by ShyamsundarR** [comment]
@raghavendrabhat if it is an ack from you, we will merge this for your VR work to proceed.

**Review by raghavendrabhat — APPROVED** [review]

File: controllers/volumereplicationgroup_controller.go (hunk 1 above)
Comment (ShyamsundarR) [reply controllers/volumereplicationgroup_controller.go:223]: Attempted to leverage the setting, but did not prove useful. Other settings for the wsl linter make it too generous. Ended up tightening another setting instead here https://github.com/RamenDR/ramen/pull/16

**Pull request merged**